src/main/bash/sdkman-utils.sh

```diff
@@ -30,20 +30,27 @@ function __sdkman_secure_curl {
 	fi
 }
 
-function __sdkman_secure_curl_download {
+function __sdkman_secure_download {
 	local curl_params="--progress-bar --location"
+	local aria2_params="-k1M -s16 -x16 -j16 --summary-interval=0"
 	if [[ "${sdkman_insecure_ssl}" == 'true' ]]; then
 		curl_params="$curl_params --insecure"
+		aria2_params="$aria2_params --check-certificate=false"
 	fi
 
 	local cookie_file="${SDKMAN_DIR}/var/cookie"
 
 	if [[ -f "$cookie_file" ]]; then
 		local cookie=$(cat "$cookie_file")
 		curl_params="$curl_params --cookie $cookie"
+		aria2_params="$aria2_params --load-cookies=$cookie_file"
 	fi
 
-	curl ${curl_params} "$1"
+	if [[ "${sdkman_download_tool}" == 'aria2' ]]; then
+		aria2c --dir=$(dirname "$2") --out=$(basename "$2") $aria2_params "$1"
```

Comment:
Take a look at this:
```
#include <stdio.h>

int main(int argc, char* argv[]) {
	for (int i = 0; i < argc; ++i) {
		printf("%d: %s\n", i, argv[i]);
	}
}
```

```
∴./wordcheck $(echo "a b c")
0: ./wordcheck
1: a
2: b
3: c

∴./wordcheck "$(echo a b c)"
0: ./wordcheck
1: a b c
```
This is slightly difference. With echo it won't matter, but for other commands it might.